numpy/random/tests/test_randomstate.py

```diff
@@ -812,6 +812,10 @@ def test_dirichlet_bad_alpha(self):
         alpha = np.array([5.4e-01, -1.0e-16])
         assert_raises(ValueError, random.dirichlet, alpha)
 
+    def test_dirichlet_zero_alpha(self):
+        y = random.default_rng().dirichlet([5, 9, 0, 8])
```

Comment:
This is using the result of `default_rng()`, which is a `Generator` instance (not `RandomState`), so this test should not be in this file.  I'll fix this in the pull request I am working on for https://github.com/numpy/numpy/issues/24210.